cmd/prysmctl/testnet/generate_genesis.go

```diff
@@ -0,0 +1,246 @@
+package testnet
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+	fastssz "github.com/prysmaticlabs/fastssz"
+	"github.com/prysmaticlabs/prysm/v3/cmd"
+	"github.com/prysmaticlabs/prysm/v3/config/params"
+	"github.com/prysmaticlabs/prysm/v3/io/file"
+	ethpb "github.com/prysmaticlabs/prysm/v3/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v3/runtime/interop"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+)
+
+var generateGenesisStateCmd = &cli.Command{
+	Name:   "generate-genesis",
+	Usage:  "Generate a beacon chain genesis state from specified configs",
+	Action: cliActionGenerateGenesisState,
+	Flags: []cli.Flag{
+		cmd.ChainConfigFileFlag,
+		depositJsonFileFlag,
+		configNameFlag,
+		numValidatorsFlag,
+		genesisTimeFlag,
+		outputSSZFlag,
+		outputYamlFlag,
+		outputJsonFlag,
+	},
+}
+
+var (
+	log                 = logrus.WithField("prefix", "genesis")
+	depositJsonFileFlag = &cli.StringFlag{
+		Name:  "deposit-json-file",
+		Value: "",
+		Usage: "Path to deposit_data.json file generated by the staking-deposit-cli tool for optionally specifying validators in genesis state",
+	}
+	configNameFlag = &cli.StringFlag{
+		Name:  "config-name",
+		Usage: "Config kind to be used for generating the genesis state. Default: mainnet. Options include mainnet, interop, minimal, prater, ropsten, sepolia. --chain-config-file will override this flag.",
+		Value: params.MainnetName,
+	}
+	numValidatorsFlag = &cli.Uint64Flag{
+		Name:  "num-validators",
+		Usage: "Number of validators to deterministically generate in the genesis state",
+
+		Value: 0,
+	}
+	genesisTimeFlag = &cli.Uint64Flag{
+		Name:  "genesis-time",
+		Usage: "Unix timestamp seconds used as the genesis time in the genesis state. If unset, defaults to now()",
+	}
+	outputSSZFlag = &cli.StringFlag{
+		Name:  "output-ssz",
+		Usage: "Output filename of the SSZ marshaling of the generated genesis state",
+		Value: "",
+	}
+	outputYamlFlag = &cli.StringFlag{
+		Name:  "output-yaml",
+		Usage: "Output filename of the YAML marshaling of the generated genesis state",
+		Value: "",
+	}
+	outputJsonFlag = &cli.StringFlag{
+		Name:  "output-json",
+		Usage: "Output filename of the JSON marshaling of the generated genesis state",
+		Value: "",
+	}
+)
+
+// Represents a json object of hex string and uint64 values for
+// validators on Ethereum. This file can be generated using the official eth2.0-deposit-cli.
+type depositDataJSON struct {
+	PubKey                string `json:"pubkey"`
+	Amount                uint64 `json:"amount"`
+	WithdrawalCredentials string `json:"withdrawal_credentials"`
+	DepositDataRoot       string `json:"deposit_data_root"`
+	Signature             string `json:"signature"`
+}
+
+func cliActionGenerateGenesisState(cliCtx *cli.Context) error {
+	if !cliCtx.IsSet(genesisTimeFlag.Name) {
```

Comment:
I've been trying to get away from the whole clunky `cliCtx` thing, I find just checking the value of the variable is more intuitive.

Reply:
You can also make the flag required to avoid manually validating in the command - add `Required: true` to the Flag struct.